http/src/main/java/io/micronaut/http/originatingips/IpAddressesResolver.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright 2017-2019 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.http.originatingips;
+
+import io.micronaut.http.HttpRequest;
+
+import javax.annotation.Nonnull;
+import java.util.List;
+
+/**
+ * Resolves the orginating IP Addresses for a given Request.
+ * @author Sergio del Amo
+ * @since 1.2.0
+ */
+public interface IpAddressesResolver {
```

Comment:
Should probably be called `RequestIpAddressesResolver`

Reply:
I've renamed it to `RequestIpAddressesResolver`